hw_diag/tests/test_views_diagnostics.py

```diff
@@ -86,3 +91,28 @@ def test_version_endpoint(self):
         self.assertEqual(reply['firmware_version'], '1337.13.37')
         self.assertEqual(reply['diagnostics_version'], 'aabbffe')
         self.assertEqual(reply['firmware_short_hash'], '0011223')
+
+    @patch('hw_diag.diagnostics.add_gateway_txn_diagnostic.AddGatewayTxnDiagnostic.perform_test')
+    def test_add_gateway_txn_success(self, mock):
+        endpoint = '/v1/add-gateway-txn'
+        data = dict(file=(BytesIO(), 'shipping_destination.json.asc'),)
+        resp = self.client.post(endpoint, content_type='multipart/form-data', data=data)
```

Comment:
@posterzh - This seems wrong. Because the request would contain RAW POST data, not `multipart/form-data`.

Reply:
Good catch. Thank you.